lib/compiler/test/beam_type_SUITE.erl

```diff
@@ -1433,10 +1434,22 @@ gh_7197() ->
                   ok
           end].
 
+will_succeed(_Config) ->
+    b = will_succeed_1(id(ok), id(#{})),
+    ok.
+
+%% OTP-18576: the beam_call_types:will_succeed/3 check was incorrect for 'bsl',
+%% erroneously stating that it would never fail in some instances.
+will_succeed_1(_V0, _V1)
+  when (1 bsl ((map_size(_V1) bxor 288230376151711743)
+               band 288230376151711743)) =:= _V0 ->
+    a;
+will_succeed_1(_, _) ->
+    b.
 
 %%%
 %%% Common utilities.
 %%%
-
+%%% 
```

Comment:
Seems to be an accidental change on this line.

```suggestion

```

Reply:
Thanks!